websites/A/AnimeGO/presence.ts

```diff
@@ -0,0 +1,142 @@
+				.getAttribute("content")
+				.split("/")[4]
+		}`;
+		if (
+			document.location.pathname.includes("/anime") ||
+			document.location.pathname.includes("/manga")
+		)
+			presenceData.state = document.querySelector("main h1").textContent;
+	} else if (
+		document.location.pathname === "/anime" ||
+		document.location.pathname.includes("/anime/season/") ||
+		document.location.pathname.includes("/anime/genre/") ||
+		document.location.pathname.includes("/anime/type/") ||
+		document.location.pathname.includes("/anime/status/") ||
+		document.location.pathname.includes("/anime/filter/") ||
+		document.location.pathname.includes("/anime/studio/") ||
+		document.location.pathname === "/manga" ||
+		document.location.pathname.includes("/manga/season/") ||
+		document.location.pathname.includes("/manga/genre/") ||
+		document.location.pathname.includes("/manga/type/") ||
+		document.location.pathname.includes("/manga/status/") ||
+		document.location.pathname.includes("/manga/filter/") ||
+		document.location.pathname === "/characters"
+	) {
+		presenceData.details = `В поиске ${typeCurrent}`;
+		presenceData.state = `${
+			document.querySelector(".entity-title#anime-list-title h1").textContent
+		}`;
+	} else if (
+		document.location.pathname.includes("/anime/") ||
+		document.location.pathname.includes("/manga/") ||
+		document.location.pathname.includes("/character/") ||
+		document.location.pathname.includes("/person/")
+	) {
+		const titleContent = document.querySelector(
+				`.${typeContent === "person" ? "people" : typeContent}-title h1`
+			).textContent,
+			image = document
+				.querySelector<HTMLImageElement>(
+					`.${
+						typeContent === "person" ? "people" : typeContent
+					}-poster div:nth-child(2) img`
+				)
+				.srcset.split(" ")[0];
+		presenceData.details = `Смотрит страницу ${typeCurrent}`;
+		presenceData.state = titleContent;
+		presenceData.largeImageKey = image;
+		presenceData.smallImageKey = "animego_logo";
+		presenceData.buttons = [
+			{
+				label: "Открыть страницу",
+				url: document
+					.querySelector("meta[property='og:url']")
+					.getAttribute("content")
+			}
+		];
+
+		if (video.duration !== 0) {
+			const selected = document.querySelector<HTMLSelectElement>(
+					"select[name='series']"
+				),
+				serie = `${selected.options[selected.selectedIndex].text} - ${
+					document.querySelector(
+						".episode-info-item:nth-child(2) span:nth-child(2)"
+					).textContent
+				}`;
+			presenceData.details = `Смотрит ${titleContent}`;
+			presenceData.state = serie;
+			presenceData.smallImageKey = video.paused ? "pause" : "play";
+			presenceData.smallImageText = video.paused ? strings.pause : strings.play;
+			if (video.paused) {
+				delete presenceData.startTimestamp;
+				delete presenceData.endTimestamp;
+			} else {
+				[presenceData.startTimestamp, presenceData.endTimestamp] =
+					presence.getTimestamps(
+						Math.floor(video.currentTime),
+						Math.floor(video.duration)
```

Comment:
do these really come in decimal forms? And do you really need to floor them?

Reply:
> do these really come in decimal forms? And do you really need to floor them?

Yes, without this they are not displayed.

Reply:
Can you show me an example of the data sent in `video`?

Reply:
![image](https://user-images.githubusercontent.com/62868468/167987366-54c66811-55b8-472c-8b26-67dbd893370e.png)

Reply:
It works fine for me without rounding. I don't know why it doesn't work for you.